### ISSUE TITLE
Use view and click tracking hooks on recommendations component

### ISSUE THREAD

**Issue by jroebu14** [bbc/simorgh#9081]
**Is your feature request related to a problem? Please describe.**
As an owner of the world service I want to know how links/promos perform within the Onward Journeys block so that we can get feedback on the recommendations. I also  want to know if how users are interacting with this component compared to other components on the page. 

**Describe the solution you'd like**
To track the block so we can compare it to other components on the page, add tracking to the block. This should send a view event for the block and a block click event if a link/promo is clicked. The conent of the of the view and click event for the block must be the same so ATI can calculate the click/view value.  
To track what links clicked, add tracking to the links in the Onward Journey block. This should send a view event for each link/promo in the block and a click event if a link/promo is clicked. The content of the view and click event for the link must be the same so ATI can calculate the click/view value.
This means:
view events:
A single view event for the block
A view event for each link

click events:
A click event for the block
A click event for the link

**Example**
Clicking the '"Promising Young Woman"' link on the page https://www.bbc.com/mundo/noticias-56885546

![image-2021-04-26-16-49-04-330](https://user-images.githubusercontent.com/4798332/116539044-3a593e80-a8e0-11eb-937a-d5b8d1db7740.png)

Each link has its own click and view event.

### The link click event

`atc=PUB-[Campaign]-[Creation]-[Variant]-[Format]-[General Placement]-[Detailed Placement]-[Advertiser ID]-[Url]&type=AT`

```
(atc)
atc=PUB-[cps_wsoj]-[%22Promising%20Young%20Woman%22%3A%20la%20inquietante%20realidad%20que%20retrata%20la%20pel%C3%ADcula%20nominada%20a%205%20premios%20Oscar]-[]-[CHD=promo::2]-[mundo.also_in_the_news.story.56885546.page]-[]-[mundo]-[https://www.bbc.com/mundo/noticias-56669604 ]
&type=AT
```

### The link view/impression event

`ati=PUB-[Campaign]-[Creation]-[Variant]-[Format]-[General Placement]-[Detailed Placement]-[Advertiser ID]-[Url]&type=AT`

```
(ati)
atc=PUB-[cps_wsoj]-[%22Promising%20Young%20Woman%22%3A%20la%20inquietante%20realidad%20que%20retrata%20la%20pel%C3%ADcula%20nominada%20a%205%20premios%20Oscar]-[]-[CHD=promo::2]-[mundo.also_in_the_news.story.56885546.page]-[]-[mundo]-[https://www.bbc.com/mundo/noticias-56669604 ]
&type=AT
```

### Example values from click/impression events
the values are

Campaign | cps_wsoj
-- | --
Creation | link text
Variant | empty
Format | [CHD=promo::1]
General placement | page
detailed placement | empty
advertiser ID | site of link
url | url of link

Examples:
|                    |                                                                                                                                            |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------ |
| Campaign           | cps_wsoj                                                                                                                        |
| Creation           | %22Promising%20Young%20Woman%22%3A%20la%20inquietante%20realidad%20que%20retrata%20la%20pel%C3%ADcula%20nominada%20a%205%20premios%20Oscar |
| Variant            | empty                                                                                                                                      |
| Format             | [CHD=promo::2] (Note that 2 indicates that this promo is the 2nd recommendation)                                                           |
| General Placement  | mundo.also_in_the_news.story.56885546.page                                                                                                 |
| Detailed Placement | empty                                                                                                                                      |
| Advertiser ID      | mundo                                                                                                                                      |
| Url                | https://www.bbc.com/mundo/noticias-56669604                                                                                                |

Note: To capture the relative location of the item within the block use the follow in the format string (starting with 1) CHD=promo::n

###  The Block click event
Send a block level click event if any link/promo within the block is clicked. (we are only tracking the interaction and not tracking the link location)

Don't send a click event if any other part of the block is clicked.
`atc=PUB-[Campaign]-[Creation]-[Variant]-[Format]-[General Placement]-[Detailed Placement]-[Advertiser ID]-[Url]&type=AT`

`atc=PUB-[article-sty]-[wsoj]-[]-[]-[pidgin.world.story.56829018.page]-[]-[]-[]&type=AT`

### The block view/impression event
`ati=PUB-[Campaign]-[Creation]-[Variant]-[Format]-[General]-[Detailed]-[Advertiser]-[Url]&type=AT`

`ati=PUB-[article-sty]-[wsoj]-[]-[]-[pidgin.world.story.56829018.page]-[]-[]-[]&type=AT`

### Accessibility
The solution should have no adverse affects with any of our support assistive technology. This will require a developer accessiblity swarm to check that nothing unexpected is read out, such as 'clickable'. e.g. Only the content and HTML semantics should be announced for elements, nothing extra to the defined screen reader UX for components. Note as we are using the same technical solution, this testing should be carried out on either this Issue or https://github.com/bbc/simorgh/issues/9082 - which ever is completed first.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**
The WS event tracking matrix contains all tracking campaigns and which components are appear on which page types NEWSWORLDSERVICE/WS+event+matrix.

**Comment by JonBeeb** [comment]
Updated to include details of the block level events.
